src/Catalyst.Core.Modules.Dfs.Tests/IntegrationTests/BlockExchange/BitSwapTest.cs

```diff
@@ -0,0 +1,176 @@
+#region LICENSE
+
+/**
+* Copyright (c) 2019 Catalyst Network
+*
+* This file is part of Catalyst.Node <https://github.com/catalyst-network/Catalyst.Node>
+*
+* Catalyst.Node is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System.Linq;
+using System.Text;
+using System.Threading;
+using Catalyst.Core.Lib.Dag;
+using Catalyst.Core.Modules.Dfs.BlockExchange;
+using Catalyst.Core.Modules.Hashing;
+using FluentAssertions;
+using Lib.P2P;
+using MultiFormats.Registry;
+using Xunit;
+
+namespace Catalyst.Core.Modules.Dfs.Tests.IntegrationTests.BlockExchange
+{
+    public sealed class BitSwapTest
+    {
+        private Peer self = new Peer
+        {
+            Id = "QmXK9VBxaXFuuT29AaPUTgW3jBWZ9JgLVZYdMYTHC6LLAH",
+            PublicKey =
+                "CAASXjBcMA0GCSqGSIb3DQEBAQUAA0sAMEgCQQCC5r4nQBtnd9qgjnG8fBN5+gnqIeWEIcUFUdCG4su/vrbQ1py8XGKNUBuDjkyTv25Gd3hlrtNJV3eOKZVSL8ePAgMBAAE="
+        };
+
+        public BitSwapTest()
+        {
+            new HashProvider(HashingAlgorithm.GetAlgorithmMetadata("blake2b-256"));
+        }
+
+        [Fact]
+        public void WantList()
+        {
+            var bitswap = new BitSwapService {SwarmService = new SwarmService() {LocalPeer = self}};
+            Assert.Equal(0, bitswap.PeerWants(self.Id).Count());
+
+            var cid = new DagNode(Encoding.UTF8.GetBytes("BitswapTest unknown block")).Id;
+            var cancel = new CancellationTokenSource();
+            var _ = bitswap.WantAsync(cid, self.Id, cancel.Token);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused local variable '_'.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)